testsuite/tests/backtrace/pr2195.ml

```diff
@@ -0,0 +1,18 @@
+(* TEST
+   flags += "-g"
+   ocamlrunparam = "b"
+   exit_status = "2"
+   * bytecode
+     reference = "${test_source_directory}/pr2195.byte.reference"
+   * native
+     reference = "${test_source_directory}/pr2195.opt.reference"
+     compare_programs = "false"
+*)
+
+let () =
+  Printexc.record_backtrace true;
```

Comment:
While battling with ocamltest to get this program to run with `OCAMLRUNPARAM=b` I added this line which revealed something generally interesting about the patch which hadn't been considered before. Because the debugging information is only loaded at startup if `OCAMLRUNPARAM=b` what you get with this program now is:
```
Fatal error: exception Sys_error("foo: Too many open files")
Raised by primitive operation at unknown location
Called from unknown location
```
It's probably an improvement over the fatal error, but I'm not sure if it's worth trying to improve things further?